### DISCUSSION TITLE
android webapp

### DISCUSSION THREAD

**Discussion by jghauser** [papis/papis#425] [Ideas]
@alejandrogallo's latest PR for the webapp has got me thinking. Does anyone know how easy/difficult/impossible it would be to create a self-contained webapp for android based on `papis serve`? I think this could be a realllyyy cool. What I do now on my tablet is navigate through a million folders, but it's not exactly user-friendly.

**Comment by alejandrogallo** [comment]
Me personally mostly I don't install android apps anymore, but I run them on a computer and I save a bookmark on the tablet and I browse the webapp.
Do you mean that you want to have the documents locally on your tablet
and browse them? This is certainly doable wrapping papis in electron or something like this, if someone has experience with that maybe the person can create a wrapper.

However I think having the tablet as a client is good enough and actually simpler. Of course the problem is that you have to learn how to setup
local networks and stuff like this, you could define a zerotier or tailscale network and always connect to a computer serving papis for instance.
If you don't know how to do this I could write in the near future a blog post or a documentation piece about this alternative workflow.

**Reply by jghauser** [reply]
Thanks for the feedback! I think I should definitely be able to set it up, I already have a small server for various things. But as you say, it does require quite some technical knowledge.

My problem with it though is different. I will often be travelling or outdoors and not have internet (or only very expensive roaming). I generally prefer having everything locally and synching things with syncthing (which I'm using anyway so there is little overhead). Although I guess maybe I could adjust my workflow.

Does your system play well with synchronising annotations (both those that are saved within e.g. a pdf and those, like for xournalpp, that are saved in a sidecar file in the same folder)?

**Reply by alejandrogallo** [reply]
I see, then probably having it on your android device would be the thing to do.
I think in principle it should be quite doable, there are projects like
https://python-for-android.readthedocs.io/en/latest/quickstart/ to use with a webview or simply electron. Unfortunately I can't allocate time for this, maybe someone has more experience.

I do know that some user in the past complained about the multiprocessing routines of papis because
he was running on android. https://github.com/papis/papis/pull/308
Maybe @Grimler91 kann comment if he was using termux or he actually wrapped it.

**Reply by jghauser** [reply]
Ok, thanks! I might take this on myself at some point. Maybe once the docs cleanup is done... :)
